.github/workflows/release.yml

```diff
@@ -210,7 +210,7 @@ jobs:
     name: GitHub release
     runs-on: ubuntu-18.04
     environment: build-and-publish
-    needs: [preflight]
+    needs: preflight
     if: needs.preflight.outputs.skip-publishing == 'false' || github.event.inputs.dry-run == 'true'
```

Comment:
The `dry-run` flag isn't respected. You should skip the final "Upload to OneDrive" step if it's true.